src/components/Filters.tsx

```diff
@@ -0,0 +1,81 @@
+import classNames from "classnames";
+import { ContainerType, Filters as TFilters } from "datasource";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+
+type Props = {
+  onChange: (filters: TFilters) => any;
+};
+
+/* TODO: change to keys and solve texts */
+const filtersContainerType = Object.values(ContainerType);
+
+export const Filters: React.FC<Props> = ({ onChange }) => {
+  const [activeContainerTypes, setActiveContainerTypes] = useState<
+    ContainerType[]
+  >([]);
+
+  const activeFilters = useMemo<TFilters>(
+    () => ({
+      containerTypes: activeContainerTypes
+    }),
+    [activeContainerTypes]
+  );
+
+  const handleChangeContainerTypesFilter = useCallback(
+    (value: ContainerType) => {
+      const updatedActiveContainerTypesFilter = activeContainerTypes.includes(
+        value
+      )
+        ? activeContainerTypes.filter(type => type !== value)
+        : [...activeContainerTypes, value];
+      setActiveContainerTypes(updatedActiveContainerTypesFilter);
+    },
+    [activeContainerTypes]
+  );
+
+  useEffect(() => {
+    onChange(activeFilters);
+  }, [onChange, activeFilters]);
+
+  return (
+    <div className="flex">
+      {filtersContainerType.map((filter, index) => (
+        <FilterItem<ContainerType>
+          key={`filter-containerType-${index}`}
+          className={index > 0 ? "ml-5" : ""}
+          value={filter}
```

Comment:
How do you set default filters? `FilterItem` is missing `selected: boolean`, the `checked` state doesn't reflect real state of the filter.
This will be problem when we want to load state from the URL.

Reply:
You are right.
I add solution for default filters value but found an another issues with https://github.com/OpenGov/react-leaflet-cluster-layer

Can you approve current filters? I create another issue to solve problem with clusters. Thanks.

Reply:
Fixed after chat.